war/src/main/webapp/scripts/hudson-behavior.js

```diff
@@ -2408,25 +2407,24 @@ function createSearchBox(searchURL) {
         }
 
         copyFontStyle(box,sizer);
-        copyFontStyle(box,minW);
     });
 
     // update positions and sizes of the components relevant to search
     function updatePos() {
-        function max(a,b) { if(a>b) return a; else return b; }
-
         sizer.innerHTML = box.value.escapeHTML();
-        var w = max(sizer.offsetWidth,minW.offsetWidth);
+        var w = sizer.offsetWidth;
         box.style.width =
-        comp.style.width = 
-        comp.firstChild.style.width = (w+60)+"px";
+        comp.style.width =
+        comp.firstChild.style.minWidth = (w+60)+"px";
 
         var pos = YAHOO.util.Dom.getXY(box);
         pos[1] += YAHOO.util.Dom.get(box).offsetHeight + 2;
         YAHOO.util.Dom.setXY(comp, pos);
     }
 
     updatePos();
+    // on small screens offsetWith is 0, update when screen gets bigger
+    window.addEventListener("resize", updatePos);
```

Comment:
In general, I'm very wary of resize listeners, as they can have devastating effects on performance. Have you explored other options?

If a resize listener is the way to go, I'd say we need to throttle it. One option could be using `window.requestAnimationFrame`. That would sort of ensure we can keep 60fps. It's present on all  the supported browsers, and we could just not set this listener on <IE9, as it's not essential.

We can also set the listeners just on desktop screens. We can use `window.matchMedia` for that (https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList/addListener).

Reply:
I just tried this and I think we can skip the listener altogether if we remove the first invocation of `updatePos()`. Wonder if that can have any consequence. Having min-width already does the trick.